libmcount/dynamic.c

```diff
@@ -157,73 +157,75 @@ static int find_dynamic_module(struct dl_phdr_info *info, size_t sz, void *data)
 	unsigned i;
 
 	/* TODO: support dynamic tracing for libraries */
-	if (name[0] == '\0') {
-		mdi = xzalloc(sizeof(*mdi));
+	mdi = xzalloc(sizeof(*mdi));
+
+	if (name[0] == '\0')
 		mdi->mod_name = xstrdup(read_exename());
-		mdi->base_addr = 0;
+	else
+		mdi->mod_name = xstrdup(name);
 
-		for (i = 0; i < info->dlpi_phnum; i++) {
-			if (info->dlpi_phdr[i].p_type != PT_LOAD)
-				continue;
+	for (i = 0; i < info->dlpi_phnum; i++) {
+		if (info->dlpi_phdr[i].p_type != PT_LOAD)
+			continue;
 
-			if (!base_addr_set) {
-				mdi->base_addr = info->dlpi_phdr[i].p_vaddr;
-				base_addr_set = true;
-			}
-
-			if (!(info->dlpi_phdr[i].p_flags & PF_X))
-				continue;
-
-			/* find address and size of code segment */
-			mdi->text_addr = info->dlpi_phdr[i].p_vaddr;
-			mdi->text_size = info->dlpi_phdr[i].p_memsz;
-			break;
+		if (!base_addr_set) {
+			mdi->base_addr = info->dlpi_phdr[i].p_vaddr;
+			base_addr_set = true;
 		}
-		mdi->base_addr += info->dlpi_addr;
-		mdi->text_addr += info->dlpi_addr;
 
+		if (!(info->dlpi_phdr[i].p_flags & PF_X))
+			continue;
+
+		/* find address and size of code segment */
+		mdi->text_addr = info->dlpi_phdr[i].p_vaddr;
+		mdi->text_size = info->dlpi_phdr[i].p_memsz;
+		break;
+	}
+	mdi->base_addr += info->dlpi_addr;
+	mdi->text_addr += info->dlpi_addr;
+
+	map = find_map(symtabs, mdi->base_addr);
+	if (map && map->mod) {
+		mdi->sym_base = map->start;
+		mdi->nr_symbols = map->mod->symtab.nr_sym;
+		mcount_arch_find_module(mdi, &map->mod->symtab);
 		mdi->next = mdinfo;
 		mdinfo = mdi;
-
-		map = find_map(symtabs, mdi->base_addr);
-		if (map && map->mod) {
-			mdi->sym_base = map->start;
-			mdi->nr_symbols = map->mod->symtab.nr_sym;
-			mcount_arch_find_module(mdi, &map->mod->symtab);
-		}
-
-		return 1;
 	}
 
 	return 0;
 }
 
-static int prepare_dynamic_update(struct mcount_disasm_engine *disasm,
+static void prepare_dynamic_update(struct mcount_disasm_engine *disasm,
 				  struct symtabs *symtabs)
 {
-	struct mcount_dynamic_info *mdi;
-	int ret = 0;
-
 	mcount_disasm_init(disasm);
-
 	dl_iterate_phdr(find_dynamic_module, symtabs);
+}
 
-	mdi = mdinfo;
-	while (mdi) {
-		ret = mcount_setup_trampoline(mdi);
-		if (ret < 0)
+struct mcount_dynamic_info *setup_trampoline(struct uftrace_mmap *map)
+{
+	struct mcount_dynamic_info *mdi;
+
+	for (mdi = mdinfo; mdi != NULL; mdi = mdi->next)
+	{
```

Comment:
It doesn't follow the coding style as well. The curly brace has to be at the right of the previous line.